Crash on navigating to a Web Resource

Clicking on a Web Resource, az browse crashed and provided this log

rosmith@Simian:~/src/Mercury$ azbrowse --navigate /subscriptions/Subid
AzBrowse is waiting for access to '~/.azbrowse.db', do you have another instance of azbrowse open?
Loading db complete
2019/11/07 17:19:51 Current version is the latest
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/lawrencegripper/azbrowse/pkg/endpoints.(*EndpointInfo).Match(0xc000487380, 0xc000e76100, 0x80, 0x0, 0x0)
        /go/src/github.com/lawrencegripper/azbrowse/pkg/endpoints/endpointinfo.go:115 +0x553
github.com/lawrencegripper/azbrowse/pkg/swagger.getResourceTypeForURLInner(0xc000e76100, 0x80, 0xc000d26000, 0x29c, 0x29c, 0x0)
        /go/src/github.com/lawrencegripper/azbrowse/pkg/swagger/types.go:55 +0xee
github.com/lawrencegripper/azbrowse/pkg/swagger.GetResourceTypeForURL(0x1541640, 0xc000022098, 0xc000e76100, 0x80, 0xc000d26000, 0x29c, 0x29c, 0x0)
        /go/src/github.com/lawrencegripper/azbrowse/pkg/swagger/types.go:51 +0x10f
github.com/lawrencegripper/azbrowse/internal/pkg/expanders.(*SwaggerResourceExpander).DoesExpand(0xc00000e858, 0x1541640, 0xc000022098, 0xc00068aa80, 0xc00000e3a0, 0xc0012028a0, 0xc00068aa80)
        /go/src/github.com/lawrencegripper/azbrowse/internal/pkg/expanders/swagger.go:104 +0x108
github.com/lawrencegripper/azbrowse/internal/pkg/expanders.ExpandItem(0x1541640, 0xc000022098, 0xc00068aa80, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/lawrencegripper/azbrowse/internal/pkg/expanders/expander.go:37 +0x157f
github.com/lawrencegripper/azbrowse/internal/pkg/views.(*ListWidget).ExpandCurrentSelection(0xc00099a000)
        /go/src/github.com/lawrencegripper/azbrowse/internal/pkg/views/list.go:221 +0x28d
github.com/lawrencegripper/azbrowse/internal/pkg/keybindings.ListExpandHandler.Fn.func1(0xc000594000, 0xc0001d41e0, 0xa, 0xc001153e01)
        /go/src/github.com/lawrencegripper/azbrowse/internal/pkg/keybindings/listhandlers.go:309 +0x2a
github.com/stuartleeks/gocui.(*Gui).execKeybindings(0xc000594000, 0xc0001d41e0, 0xc001153e28, 0x700000000, 0x2, 0x3)
        /go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:629 +0xc4
github.com/stuartleeks/gocui.(*Gui).onKey(0xc000594000, 0xc001153e28, 0x2, 0x0)
        /go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:593 +0x1b3
github.com/stuartleeks/gocui.(*Gui).handleEvent(...)
        /go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:413
github.com/stuartleeks/gocui.(*Gui).MainLoop(0xc000594000, 0xc000022098, 0xc000594000)
        /go/pkg/mod/github.com/stuartleeks/gocui@v0.4.1/gui.go:373 +0x1e3
main.main()
        /go/src/github.com/lawrencegripper/azbrowse/cmd/azbrowse/main.go:78 +0x1fe

--- a/pkg/endpoints/endpointinfo.go
+++ b/pkg/endpoints/endpointinfo.go
@@ -112,8 +112,10 @@ func (ei *EndpointInfo) Match(url string) MatchResult {
 		remainingURLToMatch = remainingURLToMatch[len(segment.Prefix):]
 		if segment.Match != "" {
 			// literal match - check (case-insensitively) that the remainingURL starts with segment.Match
-			matchTest := remainingURLToMatch[:len(segment.Match)]
-			if strings.EqualFold(segment.Match, matchTest) {
+			if len(segment.Match) > len(remainingURLToMatch) {
+				return MatchResult{IsMatch: false}
+			}
+			if matchTest := remainingURLToMatch[:len(segment.Match)]; strings.EqualFold(segment.Match, matchTest) {
 				remainingURLToMatch = remainingURLToMatch[len(segment.Match):]
 			} else {
 				return MatchResult{IsMatch: false}

--- a/pkg/endpoints/endpointinfo_test.go
+++ b/pkg/endpoints/endpointinfo_test.go
@@ -90,6 +90,18 @@ func TestMatchEndingWithNamedSegmentIsNotTooGreedy(t *testing.T) {
 	}
 }
 
+func TestMatchEndingSegmentLongerThanTemplate(t *testing.T) {
+
+	//should match
+	matchResult := getMatchResult(
+		"/foo/wibble",
+		"/foo/bar")
+
+	if matchResult.IsMatch {
+		t.Error("Shouldn't match")
+	}
+}
+
 func TestMatchDifferentCase(t *testing.T) {
 
 	//should match even though case differs on literal segments

